pkg/controller/dataimportcron-controller.go

```diff
@@ -218,20 +219,27 @@ func (r *DataImportCronReconciler) update(ctx context.Context, dataImportCron *c
 		now := metav1.Now()
 		dataImportCron.Status.LastExecutionTimestamp = &now
 
-		if dataVolume.Status.Phase == cdiv1.Succeeded {
+		switch dataVolume.Status.Phase {
+		case cdiv1.Succeeded:
 			importSucceeded = true
-			if err := r.updateDataSourceOnSuccess(ctx, dataImportCron); err != nil {
-				return res, err
-			}
 			if err := r.updateDataImportCronOnSuccess(ctx, dataImportCron); err != nil {
 				return res, err
 			}
+			updateDataImportCronCondition(dataImportCron, cdiv1.DataImportCronProgressing, corev1.ConditionFalse, "No current import", noImport)
 			if err := r.garbageCollectOldImports(ctx, dataImportCron); err != nil {
 				return res, err
 			}
+		case cdiv1.ImportScheduled:
+			updateDataImportCronCondition(dataImportCron, cdiv1.DataImportCronProgressing, corev1.ConditionFalse, "Import is scheduled", scheduled)
+		case cdiv1.ImportInProgress:
+			updateDataImportCronCondition(dataImportCron, cdiv1.DataImportCronProgressing, corev1.ConditionTrue, "Import is progressing", inProgress)
 		}
 	}
 
+	if err := r.updateDataSource(ctx, dataImportCron); err != nil {
```

Comment:
Looks like some new datasource management was added to this PR in addition to the conditions?

Reply:
Not much, just re-creation of DataSource if somehow deleted, and refactoring of updateDataSource so DataSource will be k8s-updated when its conditions are updated.